### DISCUSSION TITLE
Do not enforce a a 25% increase of gas premium when replacing a message

### DISCUSSION THREAD

**Discussion by dd45e640b42e6da7da96faee3996ef7c** [filecoin-project/lotus#5533] [General Q&A] [answered]
**Describe the bug**
The mpool specification (https://spec.filecoin.io/#section-systems.filecoin_blockchain.message_pool) does not specify an increase of 25% on a messages gas-premium to allow a message replace to succeed. Nonetheless `lotus mpool replace` is enforcing it.

**To Reproduce**
1. Run `lotus mpool replace ...` on a random, locally created message and do not increase the gas-premium
2. See lotus demanding a 25% increase of the gas-premium

**Expected behavior**
To not enforce values that aren't specified and therefore not needed.

**Version (run `lotus version`):**
all

**Additional context**
I get the rationale behind the 25% increase - but a miner can just choose to include the old version of the message anyways - it should be up to the user to decide how much he wants to increase the gas-premium to create an incentive for the miner to include the new version of the message.

**Comment by jennijuju** [comment]
@Kubuxu i think you may have thoughts here?

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
```
lotus mpool config
{"PriorityAddrs":null,"SizeLimitHigh":30000,"SizeLimitLow":20000,"ReplaceByFeeRatio":1.25,
"PruneCooldown":60000000000,"GasLimitOverestimation":1.25}

lotus mpool config --help
NAME:
   lotus mpool config - get or set current mpool configuration

USAGE:
   lotus mpool config [command options] [new-config]

OPTIONS:
   --help, -h  show help (default: false)
```
[line break added for easier reading]

one can obviously change the settings here - i think. Now the question is if the `ReplaceByFeeRatio":1.25` is enforced when receiving a message another peer send or not.

if it is then there should be something in the docs to document this behavior. the specs do not imply it.

[edit]
is `ReplaceByFeeRatio":1.25` actually the value that is responsible for the 25% increase enforcement of the gas-premium in the replace command? 

i am not sure right now if this is really just some kind of anti spam measure or if i miss something.

**Comment by hyunmoon** [comment]
**Q:** Is the 25% increase in GasPremium for replacing a message required by the chain or the lotus client?
**A:** It is required by mpool, chain doesn't know about other messages. The reason why it is there is so it makes sense to miners which message is the best (gives them highest reward).

**Q**: How about lifting the +25% GasPremium requirement for replacing a message? Why 25%? Can't it be 10%?
**A**: We took the 25% to perform Replace By Fee from Ethereum.

[https://filecoinproject.slack.com/archives/C0179RNEMU4/p1605901372252300?thread_ts=1605888166.231100&cid=C0179RNEMU4](url)
[https://filecoinproject.slack.com/archives/C0179RNEMU4/p1607129594241000?thread_ts=1607103246.206400&cid=C0179RNEMU4](url)

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
thanks @hyunmoon 

so we just need to properly document this behavior. an alternative mpool implementation should know about that the reference implementation is setting this!

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment, marked as answer]
this is a very strange solution for a problem - is this solving a problem? not sure... but the argument that "eth is doing it" is weak for choosing a number. an increase by 1 attoFIL has the same effect ultimately if the argument for this is "he reason why it is there is so it makes sense to miners which message is the best (gives them highest reward)"

[edit]
it's not solving the problem - it intends to give incentives to not create the problem.

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
so maybe change the title of this to:

decrease the increase of gas-premium to 1 attoFIL ;)
